sdk/servicebus/Azure.Messaging.ServiceBus/tests/Message/MessageLiveTests.cs

```diff
@@ -178,5 +179,116 @@ private class TestBody
             public int B { get; set; }
             public bool C { get; set; }
         }
+
+        [Test]
+        public async Task SendDataBodyMessageSingle()
+        {
+            await using var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false);
+
+            var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+            var sender = client.CreateSender(scope.QueueName);
+
+            var bodyBytes = new byte[] { 1, 2, 3, 4, 5 };
+            var body = new[] { new ReadOnlyMemory<byte>(bodyBytes) };
+            var msg = sender.CreateAmqpDataMessage(body);
+            Assert.AreEqual(AmqpBodyType.Data, msg.GetAmqpBodyType());
+            Assert.AreEqual(default(BinaryData), msg.TransportBody.Body);
+            var amqpTransportBody = msg.TransportBody as AmqpTransportBody;
+            Assert.NotNull(amqpTransportBody);
+            Assert.Null(amqpTransportBody.Value);
+            Assert.Null(amqpTransportBody.Sequence);
+            Assert.NotNull(amqpTransportBody.Data);
+            Assert.IsNotEmpty(amqpTransportBody.Data);
+            Assert.AreSame(body, amqpTransportBody.Data);
+
+            await sender.SendMessageAsync(msg);
+
+            var receiver = client.CreateReceiver(scope.QueueName);
+            var received = await receiver.ReceiveMessageAsync();
+            Assert.AreEqual(AmqpBodyType.Data, received.GetAmqpBodyType());
+            var receivedData = received.GetAmqpDataBody();
+            Assert.NotNull(receivedData);
+            var receivedDataArray = receivedData.ToArray();
+            Assert.IsNotEmpty(receivedDataArray);
+            var receivedDataBytes = receivedDataArray[0].ToArray();
+            Assert.AreEqual(bodyBytes.Length, receivedDataBytes.Length);
```

Comment:
Does doing an Assert.AreEqual on the entire array work here? I think this is what we do elsewhere.

Reply:
I'll test that. I'm not extremely familiar with nUnit anymore.

Reply:
AreEqual does work for arrays. I've updated the tests I already had to use this.